src/api/c/indexing_common.cpp

```diff
@@ -0,0 +1,40 @@
+/*******************************************************
+ * Copyright (c) 2018, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <cmath>
+#include <mutex>
+#include <common/err_common.hpp>
+#include <indexing_common.hpp>
+
+af_index_t spanner()
```

Comment:
spanner sounds like an action. This is just returning a span af_index_t object. This should really be a constructor of the af_index_t object or a factory function.

Reply:
Check my response to your comment near `call_once` usage. Add an class(constructor) just for this case ? I don't think it is needed. I am all ears for a new name for `spanner`, I tried to think of one but didn't find any.